app/models/ems_event/automate.rb

```diff
@@ -2,14 +2,34 @@ class EmsEvent
   module Automate
     extend ActiveSupport::Concern
 
+    def options
+      return @options if defined?(@options)
+      @options = Settings.ems_refresh
+    end
+
+    def refresher_options
+      options[ext_management_system.try(:class).try(:ems_type)]
+    end
+
+    def event_payload_refresh?
+      # When :event_targeted_refresh is true and :inventory_object_refresh is true, we will send EmsEvent object as a
+      # target, allowing the manager to do a refresh based on event payload.
+      refresher_options.try(:[], :event_targeted_refresh) && refresher_options.try(:[], :inventory_object_refresh)
+    end
+
     def refresh(*targets, sync)
       targets = targets.flatten
       return if targets.blank?
 
-      refresh_targets = targets.collect { |t| get_target("#{t}_refresh_target") unless t.blank? }.compact.uniq
-      return if refresh_targets.empty?
+      if event_payload_refresh?
+        EmsRefresh.queue_refresh(self, nil, sync)
```

Comment:
@lfu I think the idea is that the `EmsEvent` payload has enough information in it to be able to perform the refresh without requesting that data from the provider again, it wouldn't cause the provider to send the event again.